EIPS/eip-6036.md

```diff
@@ -0,0 +1,116 @@
+---
+eip: 6036
+title: Subscribeable NFT Extension
+description: 721sub is a subscribable nft protocol that allows users to enjoy their benefits for a certain period of time.
+author: Diven (@aboutmydreams), Kaso (@kasoqian), Jason (@350306878)
+discussions-to: https://github.com/ethereum/EIPs/pull/6036
+status: Draft
+type: Informational
+created: 2022-08-17
+requires: 165, 721
+---
+
+## Abstract
+
+Subscribeable NFT Extension is a standard that allows more platforms to directly use the interface to determine whether a user is still within the enjoyment time of a certain benefit
+
+## Motivation
+
+Web2 subscriptions are leased, not proprietary. You don't own a membership to your streaming audio subscription site Spotify, so you can't sell that membership. Since Apple takes a 30% commission from content providers, you're actually paying Apple's "platform tax." But this is just a preliminary idea, the following functions are what subscription NFTs can really do.
+
+
+## Specification
+
+```
+interface IERC6036 {
+    event TokenIsExpire(uint indexed tokenId);
+
+    function tokenSubscribeExtend(uint _tokenID, uint _time) external payable;
+
+    function tokenSubscribeRevoke(uint _tokenID) external;
+
+    function isTokenExpire(uint _tokenID) external returns (bool);
+
+    function queryTokenExpire(uint _tokenID) external returns(uint);
```

Comment:
How about these names (the names you have right now are grammatically incorrect):
```suggestion
    event TokenExpire(uint indexed tokenId);

    function extendTokenSubscription(uint _tokenID, uint _time) external payable;

    function revokeTokenSubscription(uint _tokenID) external;

    function hasExpired(uint _tokenID) external returns (bool);

    function tokenExpiration(uint _tokenID) external returns (uint);
```

Reply:
thanks, I think these modifications are more explicit